test/test_tensorclass.py

```diff
@@ -2937,6 +2943,65 @@ def __setattr__(self, key, value):
         assert (s.b == 2).all()
 
 
+class TestTensorOnly:
+    class TensorOnly(TensorClass["tensor_only"]):
+        a: torch.Tensor
+        b: torch.Tensor
+        c: torch.Tensor | None = None
+
+    def test_tensor_only_base(self):
+        x = self.TensorOnly(1, 2, 3)
+        assert x.a == 1
+        assert x.b == 2
+        assert x.c == 3
+        assert isinstance(x.a, torch.Tensor)
+        assert isinstance(x.b, torch.Tensor)
+        assert isinstance(x.c, torch.Tensor)
+
+    def test_tensor_only_none(self):
+        x = self.TensorOnly(1, 2)
+        assert x.a == 1
+        assert x.b == 2
+        assert x.c is None
+        x.c = 3
+        assert x.c == 3
+        assert isinstance(x.c, torch.Tensor)
+        delattr(x, "c")
+        assert not hasattr(x, "c")
+
+    def test_wrong_tensor_only(self):
+        class TensorOnly(TensorClass["tensor_only"]):
+            a: torch.IntTensor
+            b: torch.LongTensor
+            c: torch.Tensor | None = None
+            d: torch.Tensor | Union[torch.IntTensor, torch.LongTensor] | None = (
+                None  # noqa
+            )
+            e: Optional[torch.IntTensor] = None  # noqa
+            f: Optional[torch.IntTensor | None] = None  # noqa
+
+        with pytest.raises(TypeError):
```

Comment:
Do you want to capture part of the message to guarantee it's the right error being raised?
```suggestion
        with pytest.raises(TypeError, match="..."):
```